pkg/cmdlets/auth/login.go

```diff
@@ -178,3 +181,19 @@ func promptToken(credentials *mono_models.Credentials, out output.Outputer, prom
 
 	return nil
 }
+
+// AuthenticateWithDevice attempts to authenticate this device with the Platform.
+func AuthenticateWithDevice(out output.Outputer) error {
+	err := authentication.LegacyGet().AuthenticateWithDevice(func(userCode, uri string) {
```

Comment:
The callback feels unnecessary. Just have `AuthenticateWithDevice` communicate back its result and use that.

Reply:
The POST in `AuthenticateWithDevice` produces a user code and a URI that need to be communicated to the user before they can click the "Authorize" button. `AuthenticateWithDevice` is blocking on receiving the click, so the callback is used to facilitate this sequence of events.

Reply:
I have broken up the authentication process into two parts: request authorization, and wait for user action. This routine now makes use of those without a callback.